solvers/irls.py

```diff
@@ -0,0 +1,56 @@
+from benchopt import BaseSolver, safe_import_context
+from benchopt.stopping_criterion import SufficientProgressCriterion
+
+with safe_import_context() as import_ctx:
+    import numpy as np
+    from sklearn.linear_model import Ridge
+
+
+class Solver(BaseSolver):
+    name = "IRLS"  # Iterative Reweighted Least Squares
+    stopping_criterion = SufficientProgressCriterion(
+        patience=6, strategy="iteration"
+    )
+
+    references = [
+        "Y. Grandvalet, "
+        '"Least absolute shrinkage is equivalent to quadratic penalization" '
+        "International Conference on Artificial Neural Networks, "
+        "pp. 201-206, (1998)",
+    ]
+    # see also: https://homepages.laas.fr/vmagron/masmode/Gabriel.pdf
+
+    def skip(self, X, y, lmbd, fit_intercept):
+        # XXX - not implemented but not too complicated to implement
+        if fit_intercept:
+            return True, f"{self.name} does not handle fit_intercept"
+
+        return False, None
+
+    def set_objective(self, X, y, lmbd, fit_intercept):
+        self.X, self.y, self.lmbd = X, y, lmbd
+        self.fit_intercept = fit_intercept
+
+    def run(self, n_iter):
+        def w_opt(eta):
+            sqrt_eta = np.sqrt(eta)
+            ridge = Ridge(alpha=self.lmbd, fit_intercept=False).fit(
+                self.X * sqrt_eta, self.y
+            )
+            return ridge.coef_ * sqrt_eta
+
+        def eta_opt(w):
+            return np.abs(w)
+
+        n_features = self.X.shape[1]
+        eta = np.abs(self.X.T @ self.y)  # init needs to be > 0
```

Comment:
Can you briefly comment the rationale for this choice (rather than OLS for example ?)

Reply:
this is a cheap proxy for OLS without doing the computation (OLS solution if X^top X= Id).

Reply:
Does it bring noticeable improvement on a benchmark?

Reply:
nope. I simplified, and init is now `np.ones(n_features)` meaning your start in the loop with a vanilla Ridge.